project1/application.py

```diff
@@ -27,20 +15,24 @@ def index():
 
 @app.route("/register",methods=["GET","POST"])
 def register():
-    register_message =""
+    message = ""
     if request.method == "POST":
-        email=request.form.get('email') 
-        userPassword=request.form.get('password')
-        data=db.execute("SELECT email FROM users").fetchall()
-        for i in range(len(data)):
-            if data[i]["email"]==email:
-                register_message="email already exist"
-                return render_template('success.html',text=register_message)
-        db.execute("INSERT INTO users (email,password) VALUES (:a,:b)",{"a":email,"b":userPassword})
-        db.commit()
-        register_message="Success! You can log in now."
-        return render_template('success.html', text =register_message)
-
+        user_name = request.form.get("first_name")
+        email = request.form.get("email")
+        password = request.form.get("password")
+
+        data = User.query.filter_by(email=email).all()
```

Comment:
Can this be checked by catching the exception at the time of inserting the record?

Reply:
Yes, we can use an exception handler over there but I think of keeping it simple for now.

Reply:
Catching an exception is a performance advantage. You are reducing one request to the database server.

Reply:
Yeah. I have updated it.